packages/protocol/test-sol/stability/PairManager.t.sol

```diff
@@ -0,0 +1,431 @@
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        testStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    vm.expectRevert("Caller is not the Broker");
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+  }
+
+  function test_updateBuckets_whenPairDoesNotExist_shouldRevert() public {
+    changePrank(broker);
+
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        anotherTestStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    vm.expectRevert("A pair with the specified id does not exist");
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+  }
+
+  function test_updateBuckets_whenSenderIsBroker_shouldUpdateAndEmit() public {
+    changePrank(broker);
+
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        testStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    PairManager.Pair memory pairBeforeUpdate = testee.getPair(pairId);
+    assert(pairBeforeUpdate.stableBucket == 250000000000);
+    assert(pairBeforeUpdate.collateralBucket == 250000000000);
+
+    vm.expectEmit(false, false, false, false);
+    emit BucketsUpdated(pairId, 45000000000, 45000000000);
+    testee.updateBuckets(pairId, 45000000000, 45000000000);
+
+    PairManager.Pair memory pairAfterUpdate = testee.getPair(pairId);
+    assert(pairAfterUpdate.stableBucket == 45000000000);
+    assert(pairAfterUpdate.collateralBucket == 45000000000);
+  }
+
+  /* ---------- Getters ---------- */
+
+  function testFail_getPair_whenPairDoesNotExist_shouldRevert() public {
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        anotherTestStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    // vm.expectRevert("A pair with the specified id does not exist");
+    testee.getPair(pairId);
+  }
+
+  function test_getPair_whenPairExists_shouldReturnPair() public {
+    bytes32 pairId = keccak256(
+      abi.encodePacked(
+        testStableAsset.symbol(),
+        testCollateralAsset.symbol(),
+        testMentoExchange.name()
+      )
+    );
+
+    PairManager.Pair memory existingPair = testee.getPair(pairId);
+    assert(existingPair.stableAsset == address(testStableAsset));
+    assert(existingPair.collateralAsset == address(testCollateralAsset));
+    assert(existingPair.mentoExchange == testMentoExchange);
+    assert(existingPair.collateralBucketFraction.equals(FixidityLib.wrap(0.1 * 10**24)));
+    assert(existingPair.stableBucketMaxFraction.equals(FixidityLib.wrap(0.1 * 10**24)));
+    assert(existingPair.spread.equals(FixidityLib.wrap(0.1 * 10**24)));
+  }
+}
+
+contract PairManagerTest_createPair is PairManagerTest {
+  function test_createPair_whenNotCalledByOwner_shouldRevert() public {
```

Comment:
I know you said you want the method name in the test but the reason I went with different contracts for test is to sort of mimic `describe` blocks. So having `_createPair` in the contract name makes me want to skip it from the test name as those can get out of hand, but I'm happy to do both.

Reply:
I understand what you were going for. My thinking behind the naming was for debugging failures easily in CI builds or locally. In the logs the test name is more prominent than the file name especially when you have a lot of tests, so I think it makes sense to have more info in the test name. I don't mind keeping both or removing the function from the filename.